frame/support/procedural/src/construct_runtime/expand/origin.rs

```diff
@@ -160,7 +179,7 @@ pub fn expand_outer_origin(
 				match self.caller {
 					// Root bypasses all filters
 					OriginCaller::system(#system_path::Origin::<#runtime>::Root) => true,
-					_ => (self.filter)(call),
+					_ => self.named_filters.values().all(|f| (*f)(call)) && (*self.filter)(call),
```

Comment:
i assume it is not really possible to overload the number of filters, making this logic heavy?